src/azul/chalice.py

```diff
@@ -93,6 +95,12 @@ def _authentication_middleware(self, event, get_response):
             response = get_response(event)
         return response
 
+    def _lambda_context_middleware(self, event, get_response):
+        if not self.unit_test:
+            remaining_millis = self.lambda_context.get_remaining_time_in_millis()
+            config.is_api_gateway_context = remaining_millis < 30000
```

Comment:
Here is the lambda bootstrap.py that populates the Lambda context; https://gitlab.dev.singlecell.gi.ucsc.edu/-/snippets/30.

Take a look at `class LambdaContext` in that file, it looks like there is a
`epoch_deadline_time_in_ms` passed in from the service that invoked the Lambda. 

I tested this out with a small log statement to observe the remaining time when invoking the lambda from APIGateway and having the manifest step function hit invoke the function. In the former I saw that the remaining time was 30999 ms, but the latter causes a python error: 

```
<span>

@ingestionTime | 1647370933235
-- | --
@log | 122796619775:/aws/lambda/azul-service-amar-manifest
@logStream | 2022/03/15/[$LATEST]ac07f238917a46cfb68f437496bb8309
@message | [ERROR] AttributeError: 'NoneType' object has no attribute 'get_remaining_time_in_millis'Traceback (most recent call last):  File "/var/task/chalice/app.py", line 1564, in __call__    return self.handler(event_obj)  File "/var/task/chalice/app.py", line 1526, in __call__    return self.handler(request, self.next_handler)  File "/var/task/azul/chalice.py", line 100, in _lambda_context_middleware    remaining_millis = self.lambda_context.get_remaining_time_in_millis()
@timestamp | 1647370924358

</span>
```

Here was the patch applied to your branch:
```
Index: src/azul/chalice.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/chalice.py b/src/azul/chalice.py
--- a/src/azul/chalice.py	(revision b00ee2a79c5ed87f871a0e0294a0023cd1a793d5)
+++ b/src/azul/chalice.py	(date 1647370539078)
@@ -252,6 +252,8 @@
             log.info('Authenticated request as %r', auth)
 
     def _log_request(self):
+        log.debug('Time left: %s', self.lambda_context.get_remaining_time_in_millis())
+        log.debug('Invoked Function ARN: %s', self.lambda_context.invoked_function_arn)
         if log.isEnabledFor(logging.INFO):
             context = self.current_request.context
             query = self.current_request.query_params
```

During your tests were you able to get this to pass IT?

Reply:
Hmm, it seemed to pass yesterday but now it's not. I'm guessing I made a mistake trying to test it (deployed the wrong branch etc). Thanks for being thorough.